tests/FnStreamTest.php

```diff
@@ -88,4 +88,12 @@ public function testDecoratesWithCustomizations()
         $this->assertEquals('foo', $b->read(3));
         $this->assertTrue($called);
     }
+
+    public function testDoNotAllowUnserialization()
+    {
+        $this->setExpectedException('\LogicException');
```

Comment:
this should be after the constructor. also please test the exception message for clarity